sdks/python/apache_beam/runners/interactive/recording_manager.py

```diff
@@ -454,8 +455,12 @@ def record(self, pcolls, max_n, max_duration):
           'options is deprecated since First stable release. References to '
           '<pipeline>.options will not be supported',
           category=DeprecationWarning)
-      pf.PipelineFragment(list(uncomputed_pcolls),
-                          self.user_pipeline.options).run(blocking=True)
+      from apache_beam.runners.portability.flink_runner import FlinkRunner
+      flink_runner_used = isinstance(
```

Comment:
You may make this `is_remote_run` so that the blocking is more self-explained. Also, you can check if the cache root of the current pipeline's cache manager is on GCS instead of checking the runner type: so no matter what runner the run uses, as long as you need to wait for the file on GCS, you run it blocking.

Reply:
Sounds good, I've incorporated this change now.